### ISSUE TITLE
[deprecated_message] results in a lua exception

### ISSUE THREAD

**Issue by Pentarctagon** [wesnoth/wesnoth#2425]
The following results in `<Lua caught unknown exception>` being printed on the screen:
```
[event]
  name=start

  [deprecated_message]
      logger=warn
      message=_"Tabib is an obsolete name. Use Dune Apothecary instead."
  [/deprecated_message]
[/event]
```

This is related to the [Dunefolk unit renaming](https://forums.wesnoth.org/viewtopic.php?f=12&p=623003#p623003).

**Comment by Vultraz** [comment]
It seems @CelticMinstrel forgot to commit the implementation of `wesnoth.deprecation_message` in c734f569227305bbae846facb9822f0eefe3c125...

**Comment by CelticMinstrel** [comment]
What are you talking about? It's right here: https://github.com/wesnoth/wesnoth/blob/c734f569227305bbae846facb9822f0eefe3c125/data/lua/core.lua#L153

Though it shouldn't be local, mind you. Still, I doubt that actually makes a difference when it's being defined in a global table. Glancing over the code, the probable cause for your Lua error is actually the missing `level` attribute. Using an invalid level _is_ supposed to be an error, but I'm not sure why the error message isn't being printed onscreen...

You're using the tag wrong, in any case. Maybe the documentation needs updating? Anyway, you want it to look more like this (as seen in _deprecation-utils.cfg_):

```
[deprecation_message]
    what=Tabib
    level=3
    message=_"This unit has been renamed to:" + " " + "Dune Apothecary"
    version=1.14
[/deprecation_message]
```

The `logger` key is not supported (the choice of logger is based on the `level` key instead; level 3 uses the error logger, level 2 uses the warning logger, but I think level 3 is the desired level here). I'm pretty sure it's _never_ written to the WML log (thus dodging any complaints Zookeeper might have about error messages being shown to all users). I do want it to go to the WML log, but I think there were other things that needed to be done first to allow deprecation messages to be shown only to people who want to see them.

In the above example I've attempted to arrange the message so that it adds only a single translated string. It's perhaps not ideal, as it may lead to unnatural word order in other languages, but that's also why it ends with a colon. An alternative would be to use variables (since this is in an action): `_"This unit has been renamed to $deprecated_unit_new_name."` In that case obviously you'd add a `[set_variable]` and `[clear_variable]` tag before and after to populate the required variable.

**Comment by CelticMinstrel** [comment]
Fixed in b4b54405a9147c16fa7ac691f0a5619ade79a46e.
